connector/connect/common/src/main/protobuf/spark/connect/commands.proto

```diff
@@ -177,3 +179,118 @@ message WriteOperationV2 {
   // (Optional) A condition for overwrite saving mode
   Expression overwrite_condition = 8;
 }
+
+// Starts write stream operation as streaming query. Query ID and Run ID of the streaming
+// query are returned.
+message WriteStreamOperationStart {
+
+  // (Required) The output of the `input` streaming relation will be written.
+  Relation input = 1;
+
+  // The following fields directly map to API for DataStreamWriter().
+  // Consult API documentation unless explicitly documented here.
+
+  string format = 2;
+  map<string, string> options = 3;
+  repeated string partitioning_column_names = 4;
+
+  oneof trigger {
+    string processing_time_interval = 5;
+    bool available_now = 6;
+    bool one_time = 7;
+    string continuous_checkpoint_interval = 8;
+  }
+
+  string output_mode = 9;
+  string query_name = 10;
+
+  // The destination is optional. When set, it can be a path or a table name.
+  oneof sink_destination {
+    string path = 11;
+    string table_name = 12;
+  }
+}
+
+message WriteStreamOperationStartResult {
+
+  // (Required)
+  string query_id = 1;
```

Comment:
I still think we should add more documentation here. Proto is an API and developers read it. Right now I do not know what is a `query_id` and a `run_id` when reading these.

I think a good example is what Beam document their proto: https://github.com/apache/beam/blob/master/model/fn-execution/src/main/proto/org/apache/beam/model/fn_execution/v1/beam_fn_api.proto

Reply:
@amaliujia let's not repeat the existing Spark documentation here. 
I mentioned multiple times in the is file that we should look at corresponding documentation for the API method. 
In addition, lets have more thorough example of good documentation in spark-connect. I don't think we should start that that with this PR.

Reply:
It's ok that we do not have it at the beginning. But I do hope eventually we could improve.

Reply:
Updated. Yes, lets have uniform standard across spark-connect and update documentation in the proto files. 

Note that this is just the beginning. 90% streaming work still left. These protos will mostly change.